app/assets/src/components/ui/controls/dropdowns/ThresholdFilterDropdown.jsx

```diff
@@ -23,12 +23,42 @@ class ThresholdFilterDropdown extends React.Component {
     let newThresholds = props.thresholds.filter(
       ThresholdFilterDropdown.isThresholdValid
     );
-    if (newThresholds !== state.thresholds) {
-      return { thresholds: newThresholds };
+    if (
+      !ThresholdFilterDropdown.areThresholdsFiltersEqual(
+        newThresholds,
+        state.oldThresholds
+      )
+    ) {
+      return {
+        thresholds: newThresholds,
+        oldThresholds: newThresholds
```

Comment:
why do we need to return oldThresholds?

Reply:
Its a consequence of getDerivedPropsFromState being static. Does not have access to the old props. So the advised way is to keep track of the old props.
You should only check if the old props changed as state might change within the component.
- React's advice: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#alternative-1-reset-uncontrolled-component-with-an-id-prop
- Why old props are not included: https://github.com/facebook/react/issues/12188